Added specification for Eye-9 (#11104)
openvinotoolkit/openvino: docs/ops/generation/EyeLike_9.md

Example 1. *EyeLike* output with `output_type` = f32:

``` 
num_rows = 3

output  = [[1. 0. 0.]
           [0. 1. 0.]
           [0. 0. 1.]]
```

Example 2. *EyeLike* output with `output_type` = i32, `diagonal_index` = 2:

``` 
num_rows = 3

num_columns = 4

output  = [[0 0 1 0]
           [0 0 0 1]
           [0 0 0 0]]
```

Example 3. *EyeLike* output with `output_type` = i32, `diagonal_index` = -1:

``` 
num_rows = 3

num_columns = 4

output  = [[0 0 0 0]
           [1 0 0 0]
           [0 1 0 0]]
```

Example 4. *EyeLike* output with `output_type` = f16, `diagonal_index` = 5:

``` 
num_rows = 2

output  = [[0. 0.]
           [0. 0.]]
```

**Attributes**:

* *output_type*

    * **Description**: the type of the output
    * **Range of values**: any numeric type
    * **Type**: string
    * **Required**: *Yes*

* *diagonal_index*

    * **Description**: index of the diagonal to be populated. A positive value refers to an upper diagonal and a negative value refers to a lower diagonal. Value `0` populates the main diagonal.

>     * **Range of values**: any integer value

Reviewer: Is it really any integer value? If |diagonal_index| > max(num_rows, num_columns) then all elements of output matrix are zeroes?

Author: Yes! There is an example but I think this corner case should be written here. Done. Thanks!

    * **Type**: int
    * **Default value**: *0*
    * **Required**: *No*

**Inputs**:

*   **1**: `num_rows` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. **Required.**

*   **2**: `num_columns` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. Optionally, with default value equals to `num_rows`.

**Outputs**:

* **1**: A tensor with type specified by the attribute *output_type*. The shape is `[num_rows, num_columns]`

**Types**

* *T_NUM*: `int32` or `int64`.

**Examples**

*Example 1*

```xml
<layer ... name="EyeLike" type="EyeLike">
    <data output_type="i32"/>
    <input>
        <port id="0" precision="I32"/>  <!-- num rows -->
    </input>
    <output>
        <port id="3" precision="I32" names="EyeLike:0">

>             <dim>-1</dim>

Reviewer: I think it is better to add example with static output shape. Also, is this operation really can have dynamic output shape?

Author: Just imagine that inputs are constants we can write static output shape, I agree. Done

As for really dynamic outputs. 
If inputs (`row_number` and `column_number`) are parameters, output matrix shape will be dynamic.
For example, TF has `row_number` and `column_number` as parameters. So in TF this node can has dynamic output shape.
Also in ONNX this op has input - 2D matrix that can be dynamic. So we can say that ONNX support dynamic  `row_number` and `column_number` (because it's size of dynamic input shape).

Reviewer: Ok, thanks for explanation! I think it is important moment and we should mention it in out specification and have related example.

Can one dimension be dynamic and one - static?

Author: Added new examples with dynamic output, thanks!

## EyeLike <a name="EyeLike"></a> {#openvino_docs_ops_generation_EyeLike_9}

> **Versioned name**: *EyeLike-8*

Reviewer 2: ```suggestion
**Versioned name**: *EyeLike-9*
```

## EyeLike <a name="EyeLike"></a> {#openvino_docs_ops_generation_EyeLike_9}

> **Versioned name**: *EyeLike-9*

Reviewer 3: Please let us call this operation as `Eye` because most of frameworks (TF, PyTorch, MXNet) have this name.
`EyeLike` is from ONNX and uses a shape of input tensor (a bit different semantic).

**Category**: *Generation*

> **Short description**: *EyeLike* operation generates shift matrix.

Reviewer 4: ```suggestion
**Short description**: The *EyeLike* operation will generate shift matrices.
```

Author: I see that almost all specifications have a template of a short description :
`*NAME* operation generates` without _Future Simple_ and article _the_.
Moreover one operation generates one matrix not several matrices. It's important
So I'd rather leave it as it is.
Thanks!

> **Short description**: *EyeLike* operation generates shift matrix.

Reviewer 3: I think we need to mention that it generates identity matrix or ones on the diagonal

> **Short description**: *EyeLike* operation generates identity matrix.

Reviewer 5: Shift matrix ?
https://en.wikipedia.org/wiki/Shift_matrix
https://en.wikipedia.org/wiki/Identity_matrix

Author: Good catch! Thanks!

**Detailed description**:

> 

Reviewer 3: I think we need to generate a batch of identity matrices as well as TensorFlow: https://www.tensorflow.org/api_docs/python/tf/eye

Author: Done, thanks!

> *EyeLike* operation generates matrix with ones on the diagonal and zeros everywhere else. Index of the diagonal to be populated with ones is given by `diagonal_index`: `output[i, i + diagonal_index] = 1`.

Reviewer 4: ```suggestion
An *EyeLike* operation generates a matrix with ones on the diagonal and zeros everywhere else. The index of the diagonal to be populated with ones is given by `diagonal_index`: `output[i, i + diagonal_index] = 1`.
```

Author: Done, thanks

Example 1. *EyeLike* output with `output_type` = f32:

``` 
num_rows = 3

output  = [[1. 0. 0.]
           [0. 1. 0.]
           [0. 0. 1.]]
```

Example 2. *EyeLike* output with `output_type` = i32, `diagonal_index` = 2:

``` 
num_rows = 3

num_columns = 4

output  = [[0 0 1 0]
           [0 0 0 1]
           [0 0 0 0]]
```

Example 3. *EyeLike* output with `output_type` = i32, `diagonal_index` = -1:

``` 
num_rows = 3

num_columns = 4

output  = [[0 0 0 0]
           [1 0 0 0]
           [0 1 0 0]]
```

Example 4. *EyeLike* output with `output_type` = f16, `diagonal_index` = 5:

``` 
num_rows = 2

output  = [[0. 0.]
           [0. 0.]]
```

**Attributes**:

* *output_type*

    * **Description**: the type of the output
    * **Range of values**: any numeric type
    * **Type**: string
    * **Required**: *Yes*

> * *diagonal_index*

Reviewer 3: let us make this as an input. It will be more flexible and it doesn't influence on output shape

>     * **Description**: index of the diagonal to be populated. A positive value refers to an upper diagonal and a negative value refers to a lower diagonal. Value `0` populates the main diagonal. If `diagonal_index` is positive value and is not less than `num_rows` or if `diagonal_index` is negative value and is not more than `num_columns`, matrix will be filled with only zeros.

Reviewer 4: ### 
```suggestion
    * **Description**: The index of the diagonal to be populated. A positive value refers to an upper diagonal and a negative value refers to a lower diagonal. Value `0` populates the main diagonal. If `diagonal_index` is a positive value and is not smaller than `num_rows` or if `diagonal_index` is a negative value and is not larger than `num_columns`, the matrix will be filled with only zeros.
```

Author: Done, thanks!

    * **Range of values**: any integer value
    * **Type**: int
    * **Default value**: *0*
    * **Required**: *No*

**Inputs**:

*   **1**: `num_rows` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. **Required.**

> *   **2**: `num_columns` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. Optionally, with default value equals to `num_rows`.

Reviewer 4: ```suggestion
*   **2**: `num_columns` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. This attribute is optional, and its default value equals to `num_rows`.
```

Author: It's not an attribute but it's an input. It's important so I have renamed as input
Thanks!

> *   **2**: `num_columns` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. This input is optional, and its default value equals to `num_rows`.

Reviewer 5: number of columns ?

Author: Good catch! Thanks

> 

Reviewer 3: please add one more input `batch_shape` as in TF

Author: We discussed offline, done.

**Outputs**:

> * **1**: A tensor with type specified by the attribute *output_type*. The shape is `[num_rows, num_columns]`

Reviewer 4: ### 
```suggestion
* **1**: A tensor with the type specified by the *output_type* attribute. The shape is `[num_rows, num_columns]`
```
